### DISCUSSION TITLE
Complete Docker Solution

### DISCUSSION THREAD

**Discussion by mediacowboy** [openaudible/openaudible#929] [Ideas]
While I know the docker container is experimental, I could see this being an all-in-one solution for books I buy on Audible.

Run OpenAudible in docker to automatically download new books. Split the audiobook into MP3 chapters, and then have a folder structure feature that would allow something like the following. 

For Series:
Author > Series > Book {Series#} - Title > (automatic split chapters)

Non-Series:
Author > Title > (automatically split chapters)

I can work around the automatic folder structure but would still like it as it would eliminate an additional step on my end.

Automatically checking audible for new books, automatic splitting based on chapters, and not having the window pop up once done would make this perfect.

I wouldn't even mind paying extra for these features/setup.

**Comment by openaudible** [comment]
Did you try enabling the automation features under preferences? That should accomplish most of what you want (except automatic splitting and naming system..) There is also a hidden feature to automatically refresh the library every N minutes.. which you enable by editing the settings.json and you modify the autoRefreshMinutes. 

The naming system is still a work in progress... mostly done in testing.. but the edge cases are really nasty -- long book names, illegal characters in the file names, and unexpected file name collisions.. all make us hesitant to try to release something that could cause headaches for users.

**Reply by mediacowboy** [reply]
Thanks, I did not know about the autoRefreshMinues setting, and I knew most of the other stuff. Just the auto-splitting feature is all that would make this a game-changer for me. I would be happy if I could set it to auto split the MP3 to a specific folder.
